it/uk/gov/hmrc/eusubsidycompliancefrontend/services/EmailVerificationServiceSpec.scala

```diff
@@ -0,0 +1,265 @@
+
+      val undertakingCache = mock[UndertakingCache]
+
+      val escService =
+        new EscService(escConnector, undertakingCache, mock[ExchangeRateCache], mock[RemovedSubsidyRepository])
+      new EmailVerificationService(mockEmailVerificationConnector, escService, mockServicesConfig)
+    }
+
+    "approveVerificationRequest is called" must {
+
+      "report success for a successful update" in {
+        testWithRunningApp { emailVerificationService: EmailVerificationService =>
+          val expectedEmailByVerificationIdRequest =
+            ApproveEmailByVerificationIdRequest(eori = eori1, verificationId = "pending")
+
+          val verifiedEmailResponse =
+            VerifiedEmail.verifiedEmailFormat
+              .writes(VerifiedEmail(email = "", verificationId = "", verified = true))
+              .toString
+
+          escTestWireMock.approveEmailByVerificationId
+            .stubExpected(statusCode = 200, response = verifiedEmailResponse)
+
+          emailVerificationService.approveVerificationRequest(eori1, "pending").futureValue shouldBe true
+          escTestWireMock.approveEmailByVerificationId.verify(
+            expectedEmailByVerificationIdRequest
+          )
+        }
+      }
+    }
+
+    "addVerifiedEmail is called" must {
+
+      "store a new email verification request and mark it as verified" in {
+        testWithRunningApp { emailVerificationService: EmailVerificationService =>
+          val email = "foo@example.com"
+
+          val nonVerifiedEmail = VerifiedEmail(email = "", verificationId = "", verified = false)
+
+          escTestWireMock.startVerification.stubExpected(200, nonVerifiedEmail.asJson.toString)
+          escTestWireMock.approveEmailByEori.stubExpected(200, nonVerifiedEmail.copy(verified = true).asJson.toString)
+
+          emailVerificationService.addVerifiedEmail(eori4, email).futureValue shouldBe true
+
+          escTestWireMock.startVerification.verify(StartEmailVerificationRequest(eori4, email))
+          escTestWireMock.approveEmailByEori.verify(ApproveEmailAsVerifiedByEoriRequest(eori4))
+        }
+      }
+
+    }
+
+    "makeVerificationRequest is called" must {
+      val nonVerifiedEmail = VerifiedEmail(email = "", verificationId = "", verified = false)
+      val credId = "SomeAuthorityId"
+      implicit val request: AuthenticatedEnrolledRequest[AnyContent] = AuthenticatedEnrolledRequest(
+        authorityId = credId,
+        groupId = "SomeGroupId",
+        request = FakeRequest(GET, "/"),
+        eoriNumber = eori1
+      )
+
+      val email = "foo@example.com"
+
+      val createVerifyEmailRequest = EmailVerificationRequest.createVerifyEmailRequest(
+        credId,
+        "http://localhost/next-page-url",
+        email,
+        "http://localhost/previous-page-url"
+      )
+
+      "redirect to the next page if the email verification succeeded" in {
+
+        testWithRunningApp { emailVerificationService: EmailVerificationService =>
+          Mockito
```

Comment:
Could abstract the mocking calls out but it is easier to reason what is actually happening if you don't.

If you want to re-use mocking things extension methods are better than plain methods as you can actually still tell what dependencies are being operated on. Test organisation is an essay topic but they should help everyone stay fast and one of the ways to do that is to cut down bouncing around files and organise in a way that sinks into muscle memory easily.